src/App.tsx

```diff
@@ -1,14 +1,43 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | undefined>(undefined);
+  const [selectedStatus, setSelectedStatus] = useState('all');
+  const [query, setSelectedQuery] = useState('');
+
+  useEffect(() => {
+    getTodos().then(result => {
+      let filteredResult = result;
+
+      if (selectedStatus !== 'all') {
+        filteredResult = filteredResult.filter(todo =>
+          selectedStatus === 'active' ? !todo.completed : todo.completed,
+        );
+      }
+
+      if (query) {
+        filteredResult = filteredResult.filter(todo =>
+          todo.title.toLowerCase().includes(query.toLowerCase()),
+        );
+      }
+
+      setTodos(filteredResult);
+      setIsLoaded(true);
+    });
+  }, [selectedStatus, query]);
```

Comment:
Using props to initialize state directly can lead to unexpected behavior. Consider using a function inside `useState` or handling it differently to avoid potential issues.